src/blob/core.rs

```diff
@@ -323,21 +325,24 @@ where
         }))
     }
 
-    pub(crate) async fn mark_all_as_deleted(&mut self, key: &K) -> Result<Option<u64>> {
+    pub(crate) async fn mark_all_as_deleted(&mut self, key: &K) -> Result<Option<()>> {
```

Comment:
I just thought, that it is strange to return `Option<()>` to show whether the record was deleted or not. There is a better type for that: `bool`. I suggest to replace `Result<Option<()>>` to `Result<bool>`